cloud/pkg/controller/controller/downstream.go

```diff
@@ -319,44 +340,83 @@ func (dc *DownstreamController) syncEndpoints(stop chan struct{}) {
 				log.LOGGER.Warnf("Object type: %T unsupported", eps)
 				continue
 			}
+
+			ok = true
 			switch e.Type {
 			case watch.Added:
 				dc.lc.AddOrUpdateEndpoints(*eps)
 				operation = model.InsertOperation
 			case watch.Modified:
+				ok = dc.lc.IsEndpointsUpdated(*eps)
 				dc.lc.AddOrUpdateEndpoints(*eps)
 				operation = model.UpdateOperation
 			case watch.Deleted:
 				dc.lc.DeleteEndpoints(*eps)
+				dc.lc.DeleteServicePods(*eps)
 				operation = model.DeleteOperation
 			default:
 				// unsupported operation, no need to send to any node
 				log.LOGGER.Warnf("endpoints event type: %s unsupported", e.Type)
 				continue
 			}
-
 			// send to all nodes
-			dc.lc.EdgeNodes.Range(func(key interface{}, value interface{}) bool {
-				nodeName, ok := key.(string)
-				if !ok {
-					log.LOGGER.Warnf("Failed to assert key to sting")
-					return true
+			if ok {
+				var listOptions metav1.ListOptions
+				var pods *v1.PodList
+				var err error
+				svc, ok := dc.lc.GetService(fmt.Sprintf("%s/%s", eps.Namespace, eps.Name))
```

Comment:
name in this function is ```${namespace}/${name}```， ```${name}/${namespace}``` in function GetServicePods, is this correct

Reply:
Nice Catch, modified to name/namespace thanks

Reply:
Sorry, modified to namespace/name